src/vcpkg/dependencies.cpp

```diff
@@ -1484,6 +1467,16 @@ namespace vcpkg::Dependencies
                                         Versions::Scheme sb,
                                         const Versions::Version& b)
         {
+            if (sa == Versions::Scheme::Semver)
+            {
+                sa = Versions::Scheme::Relaxed;
+            }
+
+            if (sb == Versions::Scheme::Semver)
+            {
+                sb = Versions::Scheme::Relaxed;
```

Comment:
This will cause the `Versions::compare()` below to parse the semver as relaxed and therefore it won't enforce semver syntax constraints. I'm not sure if the precondition for this function is that the version texts are already syntax-checked or not.

Reply:
I'll make sure there's a test.

Reply:
I believe the existing tests (in particular the ones I already edited here) show that this is handled correctly?